File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
import java.util.*;

class Solution {
    public int solution(int n) {
        int answer = n;
        int cntN = Integer.bitCount(n);
        
        while(true)
        {
            int cnt = Integer.bitCount(++answer);
            
            if(cntN == cnt)
                break;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int count(int n)
{
    int cnt = 0;
    
    while(n != 0)
    {
        if(n % 2 == 1) 
            cnt++;
        n /= 2;
    }

    return cnt;
}

int solution(int n) {
    int cntN = count(n);
    
    while(true)
    {
        n++;

        if(count(n) == cntN)
        {
            break;
        }
    }
    
    return n;
}